File: python/Blatt4Python/src/Node.py
'''
Created on 17.06.2015

@author: ozgun
'''
class Node:
    
    
    '''
    This method generates a node.
    The data type of the parameter value is int. It's the value of the node.
    '''
    def __init__(self, value):
        
        self.value = None
        self.left = None
        self.right = None
        self.parent = None
        self.setData(value)
    '''
    The data type of the parameter value is int.
    '''
    def setData(self,value):
        self.value = value
        
    '''
    This method returns the value of the node.
    '''
    def getData(self):
        
        return self.value
    
    '''
    This method sets the left node.
    The data type of currentNode is Node. It will be set as the left node.
    '''
    def setLeft(self, currentNode):

        
            self.left = currentNode
        
        
    '''
    This method returns the left node.
    '''
    def getLeft(self):
       
        
            return self.left
    
    '''
    This method sets the right node.
    The data type of currentNode is Node. It will be set as the right node.
    '''
    def setRight(self, currentNode):
        

            self.right = currentNode        
    '''
    This method returns the right node.
    '''
    def getRight(self):
        
        
            return self.right
    
    '''
    This method sets the parent node.
    The data type of currentNode is Node. It will be set as the parent node.
    '''
    def setParent(self, currentNode):
       
        
            self.parent = currentNode
       
        
    '''
    This method returns the parent node.
    '''
    def getParent(self):
        
        if type(self.parent) is Node or self.parent is None:
            return self.parent

File: cpp/Lambda_Filtering_BinaryTree/Lambda_Filtering_BinaryTree/Node.cpp
#include <stdio.h>
#include <iostream>
#include "Node.h"


template <class T>
Node<T>::~Node() {
	delete left;
	delete right;
	delete parent;
}

template <class T>
Node<T>::Node(){

}

template <class T>
Node<T>::Node(T data){
	this->parent = nullptr;
	this->right = nullptr;
	this->left = nullptr;
	this->setData(data);
}

template <class T>
void Node<T>::setRight(Node<T>* right){
	this->right = right;
}

template <class T>
void Node<T>::setLeft(Node<T>* left){
	this->left = left;
}

template <class T>
void Node<T>::setParent(Node<T>* parent){
	this->parent = parent;
}

template <class T>
void Node<T>::setData(T data){
	this->data = data;
}

template <class T>
Node<T>* Node<T>::getRight(){
	return right;
}

template <class T>
Node<T>* Node<T>::getLeft(){
	return left;
}

template <class T>
Node<T>* Node<T>::getParent(){
	return parent;
}

template <class T>
T Node<T>::getData(){
	return data;
}